build/accelerate_launch.py

```diff
@@ -0,0 +1,111 @@
+# Copyright The FMS HF Tuning Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Script wraps launch_training to run with accelerate for multi and single GPU cases.
+Read accelerate_launch_args configuration via environment variable `SFT_TRAINER_CONFIG_JSON_PATH`
+for the path to the JSON config file with parameters or `SFT_TRAINER_CONFIG_JSON_ENV_VAR`
+for the encoded config string to parse.
+"""
+
+# Standard
+import json
+import os
+import base64
+import pickle
+import logging
+
+# Third Party
+from accelerate.commands.launch import launch_command_parser, launch_command
+import torch
+
+
+def txt_to_obj(txt):
+    base64_bytes = txt.encode("ascii")
+    message_bytes = base64.b64decode(base64_bytes)
+    try:
+        # If the bytes represent JSON string
+        return json.loads(message_bytes)
+    except UnicodeDecodeError:
+        # Otherwise the bytes are a pickled python dictionary
+        return pickle.loads(message_bytes)
+
+
+def main():
+    LOGLEVEL = os.environ.get("LOG_LEVEL", "WARNING").upper()
+    logging.basicConfig(level=LOGLEVEL)
+
+    json_configs = {}
+    json_path = os.getenv("SFT_TRAINER_CONFIG_JSON_PATH")
+    json_env_var = os.getenv("SFT_TRAINER_CONFIG_JSON_ENV_VAR")
+
+    if json_path:
+        with open(json_path, "r", encoding="utf-8") as f:
+            json_configs = json.load(f)
+
+    elif json_env_var:
+        json_configs = txt_to_obj(json_env_var)
+
+    parser = launch_command_parser()
+    # determine which flags are store true actions (don't require a value to be set)
+    actions_type_map = {
+        action.dest: type(action).__name__ for action in parser._actions
+    }
+
+    # parse accelerate_launch_args
+    accelerate_launch_args = []
+    accelerate_config = json_configs.get("accelerate_launch_args", {})
+    if accelerate_config:
+        logging.info("Using accelerate_launch_args configs: %s", accelerate_config)
+        for key, val in accelerate_config.items():
+            if actions_type_map.get(key) == "_AppendAction":
+                for param_val in val:
+                    accelerate_launch_args.extend([f"--{key}", str(param_val)])
+            else:
+                accelerate_launch_args.append(f"--{key}")
+                # Only need to add key for params that aren't flags ie. --quiet
+                if actions_type_map.get(key) == "_StoreAction":
+                    accelerate_launch_args.append(str(val))
+
```

Comment:
If a user set the ipex flag, then it would be passed as `--ipex` which would be setting it to True. But we are assuming these singular flags are always being passed, for example if the user passes in `{"ipex": false}` then we would still pass the flag through `accelerate launch --ipex` which would be setting it to true

Reply:
@tharapalanivel what do you think of this edge case, we would have to check if the flag is of store_true/store_false and then check the value given that....

Reply:
@anhuong  @tharapalanivel 

>  But we are assuming these singular flags are always being passed, for example if the user passes in {"ipex": false} then we would still pass the flag through accelerate launch --ipex which would be setting it to true


Do you think it would not create some confusion while usage? I mean for both `{"ipex": false}` and `{"ipex": true}` from the user, we are passing the flag to the `accelerate launch --ipex`, this is not factoring in the value right? 
Just trying to confirm on the thought process as this would be the interface for the users. 

@fabianlim second thoughts on this thread?

Reply:
@kmehant I think it can be written better, but maybe for now we can settle to skip the key apppend logic in the case of `_StoreTrueAction` or `StoreFalseAction`?

```python
if actions_type_map.get(key) == "_AppendAction":
   for param_val in val:
      accelerate_launch_args.extend([f"--{key}", str(param_val)])
elif not (
   (actions_type_map.get(key) == "_StoreTrueAction" and not val)
   or 
   (actions_type_map.get(key) == "_StoreFalseAction" and val)
):
   accelerate_launch_args.append(f"--{key}")
   if actions_type_map.get(key) == "_StoreAction":
      accelerate_launch_args.append(str(val))
```

Reply:
Thanks @fabianlim and @tharapalanivel 

@tharapalanivel The code snippet looks good to me, Lets go with this, thank you.

This thread resolved once @tharapalanivel pushes the changes. cc: @Ssukriti